src/map/script.cpp

```diff
@@ -14649,6 +14649,57 @@ BUILDIN_FUNC(specialeffect2)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+BUILDIN_FUNC(removespecialeffect)
+{
+	struct block_list *bl=map_id2bl(st->oid);
+	int type = script_getnum(st,2);
+	enum send_target target = script_hasdata(st,3) ? (send_target)script_getnum(st,3) : AREA;
+
+	if(bl==NULL)
+		return SCRIPT_CMD_SUCCESS;
+
+	if( type <= EF_NONE || type >= EF_MAX ){
+		ShowError( "buildin_removespecialeffect: unsupported effect id %d\n", type );
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if( script_hasdata(st,4) )
+	{
+		TBL_NPC *nd = npc_name2id(script_getstr(st,4));
+		if(nd)
+			clif_specialeffect_remove(&nd->bl, type, target);
+	}
+	else
+	{
+		if (target == SELF) {
+			TBL_PC *sd;
+			if (script_rid2sd(sd))
+				clif_specialeffect_remove_single(bl,type,&sd->bl);
+		} else {
+			clif_specialeffect_remove(bl, type, target);
+		}
+	}
+	return SCRIPT_CMD_SUCCESS;
+}
+
+BUILDIN_FUNC(removespecialeffect2)
+{
+	TBL_PC *sd;
+
+	if( script_nick2sd(4,sd) ){
```

Comment:
Please change this to cancel early (by inverting the if) and return SCRIPT_CMD_FAILURE.